uber/models.py

```diff
@@ -1182,14 +1182,14 @@ def last_first(cls):
     @property
     def watchlist_guess(self):
         try:
-            with sa.Session() as session:
-                return session.guess_attendee_watchentry(self)
+            with Session() as session:
+                return [u.__dict__ for u in session.guess_attendee_watchentry(self)]
```

Comment:
> to_dict() just plain doesn't work here

Interesting, I'll try to find some time tonight after work to check out this branch again to see why, because it really should work.

> Also, ``__dict__`` is an internal sql alchemy property, so it seems more appropriate to stick with that for compatibility. I don't want an sql alchemy update to break this because we used something that's not from sql alchemy.

Strictly speaking, ``__dict__`` is a Python property which exists on every[1] Python object and isn't specific to SQLAlchemy.  It's a dictionary which represents the internal state of (in this case) a class instance.  In general it will contain everything for the class instance which was set on the instance itself rather than being set on the class[2].

There are two problems with returning the ``.__dict__`` of an instance in order to get its values:

1) You won't get properties.  This might not matter, and I wouldn't expect it to matter in this case, since in this particular case I don't think we want anything other than the raw column values, so we can probably ignore this concern here.

2) It's possible that you'll end up with some of the object's internal state which was not intended to survive the destruction of the object.  Doing this makes it MORE likely that (as you say) "an sql alchemy update to break this", because we can't guarantee that SQLAlchemy isn't putting anything into it's instances' ``__dict__``s which aren't going to mess things up if they survive the instance having it's ``__del__`` called.

As for compatibility, ``.to_dict()`` is a method which is guaranteed (unless there's some weird bug that I don't know about, which is certainly possible) to be on every ``MagModel`` instance.  It's something that's added automatically by the ``@declarative_base`` decorator we use in ``uber/models.py`` to define our ``MagModel`` base class.

I'll try to find some time to take a look at this tonight to figure out why ``.to_dict()`` isn't working.  To point me in the right direction, what do you mean "just plain doesn't work here"?  Were you getting an exception, or was it just not returning the values you needed?


[1] Technically not ANY object because there's something called the ``__slots__`` API which is used for high-performance Python code where you can define objects without a ``__dict__`` but I've literally never used that ever.

[2] Technically there are also some situations where this isn't true, but as with my previous footnote those edge cases are usually too edge-casey to worry about.